File: AngularApp/src/app/shared/models/package.ts
export interface Package {
    codeString: string;
    dllBytes: string;
    pdbBytes: string;
    id: string;
    committedByAlias: string;
}


export interface DetectorCommit {
    sha: string;
    author: string;
    date: string;
    previousSha: string;
    onClick: Function;
    isSelected: boolean;

    // constructor(sha: string, author: string, date: string, previousSha: string, onClick: Function = null) {
    //   this.sha = sha;
    //   this.author = author;
    //   this.date = date;
    //   this.previousSha = previousSha;
    //   this.onClick = onClick;
    // }

    // constructor(label: string, onClick: Function, isSelected: Function, icon: string = null, expanded: boolean = false, subItems: CollapsibleMenuItem[] = []) {
    //     this.label = label;
    //     this.onClick = onClick;
    //     this.expanded = expanded;
    //     this.subItems = subItems;
    //     this.isSelected = isSelected;
    //     this.icon = icon;
    //   }

}

File: ApplensBackend/Models/Package.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AppLensV3.Models
{
    public class Package
    {
        public string CodeString;
        public string DllBytes;
        public string PdbBytes;
        public string Id;
    }

    public class DetectorCommit
    {
        public string Sha;
        public string Author;
        public string DateTime;
        public string PreviousSha;

        public DetectorCommit(string sha, string author, string dateTime, String previousSha)
        {
            Sha = sha;
            Author = author;
            DateTime = dateTime;
            PreviousSha = previousSha;
        }
    }
}
